cpp/src/arrow/acero/asof_join_node_test.cc

```diff
@@ -1678,5 +1678,55 @@ TEST(AsofJoinTest, BackpressureWithBatchesGen) {
                           /*slow_r0=*/false);
 }
 
+TEST(AsofJoinTest, GH40675) {
+  auto left_batch = ExecBatchFromJSON(
```

Comment:
It's nice to give the GH issue reference, but it would even be a bit better with a comment explaining what specifically this test is doing (without having to decipher all the test code).

Reply:
Addressed.

Reply:
Addressed.